lib/service/src/main/java/io/atlasmap/service/MappingService.java

```diff
@@ -91,6 +97,126 @@ public MappingService(AtlasService parent) {
         this.previewContext = this.atlasService.getContextFactory().createPreviewContext();
     }
 
+    /**
+     * Return the field from the specified list of mapped fields whose index property matches the
+     * specified field index.
+     *
+     * @param mappedFields
+     * @param fieldIndex
+     *
+     * @return field
+     */
+    private Field getFieldByIndex(List<Field> mappedFields, int fieldIndex) {
+        for (Field field : mappedFields) {
+            if (field.getIndex().intValue() == fieldIndex) {
+                return field;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Return the ordinal position of the mapped field with the specified index property.
+     *
+     * @param mappedFields
+     * @param fieldIndex
+     *
+     * @return ordinal position
+     */
+    private int getOrdinalPosition(List<Field> mappedFields, int fieldIndex) {
+        int ordinalPosition = 0;
+        for (Field field : mappedFields) {
+            if (field.getIndex().intValue() == fieldIndex) {
+                break;
+            }
+            ordinalPosition++;
+        }
+        return ordinalPosition;
+    }
+
+    /**
+     * Return the mapping element for the specified mapping uuid.
+     *
+     * @param uuid
+     * @returns mapping element
+     */
+    private Mapping getMappingByID(List<BaseMapping> mappings, String uuid) {
+        Mapping objectMapping = null;
+        if (mappings != null) {
+            for (ListIterator<BaseMapping> iter = mappings.listIterator(); iter.hasNext();) {
+                Mapping element = (Mapping)iter.next();
+                if (element.getId().equals(uuid)) {
+                    objectMapping = element;
+                    break;
+                }
+            }
+        }
+        return objectMapping;
+    }
+
+    /**
+     * Change the value of the specified current index to the specified target index for the specified
+     * mapped field (source/target).
+     *
+     * @param mappingDefinitionId
+     * @param mappingId
+     * @param dataSourceType
+     * @param currentFieldIndex
+     * @param targetFieldIndex
+     *
+     * @return empty response
+     */
+    @PUT
+    @Path("/{mappingId}/field/{dataSourceType}/current/{currentFieldIndex}/targetFieldIndex")
+    @Produces({MediaType.TEXT_PLAIN})
+    @Operation(summary = "Change Mapped Field Index", description = "Change the mapped field's index value")
+    @ApiResponses(
+        @ApiResponse(responseCode = "200", description = "The mapped field formerly at index {currentFieldIndex} is now at {targetFieldIndex}."))
+    public Response changeMappedFieldIndex(
+        @Parameter(description = "Mapping Definition ID") @PathParam("mappingDefinitionId") Integer mappingDefinitionId,
+        @Parameter(description = "Mapping ID") @PathParam("mappingId") String mappingId,
+        @Parameter(description = "Source or Target Mapping") @PathParam("dataSourceType") String dataSourceType,
```

Comment:
`DataSourceType` enum should work for the type of this parameter, to avoid String comparison at L190.